### PR TITLE
Network

### PR DESCRIPTION
**Pull request by PWiley**

### CODE REVIEW

File: SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                                 data: Data,
+                                 completion: @escaping (Result<T, Error>) -> Void) where T: Codable {
+          guard let decodedData = try? decoder.decode(type.self, from: data) else {
+              print("Unable to decode: \(type.self)")
+              completion(.failure(NetworkError.unableToDecodeData))
+              return
+          }
+          completion(.success(decodedData))
+      }
+
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift:109]: Useless line

File: SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+enum NetworkError: Error {
+    case unableToDecodeData
+    case dataAbsent
+}
+
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift:116]: Useless line

File: SoundsBinder/SoundsBinder/Sources/Network/HTTPEngine.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+//  HTTPEngine.swift
+//  SoundsBinder
+//
+//  Created by Patrick Wiley on 21.12.21.
+//
+
+
+import Foundation
+
+//typealias HTTPCompletionHandler = (Data?, HTTPURLResponse?, Error?) -> Void
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/HTTPEngine.swift:11]: Avoid keeping dead code

File: SoundsBinder/SoundsBinder/Sources/Network/RequestCancellationToken.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
+//
+//  RequestCancellationToken.swift
+//  SoundsBinder
+//
+//  Created by Patrick Wiley on 21.12.21.
+//
+
+
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/RequestCancellationToken.swift:8]: Useless line

File: SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift [hunk 5]
```diff
@@ -0,0 +1,75 @@
+//
+//  URLBuilder.swift
+//  SoundsBinder
+//
+//  Created by Patrick Wiley on 21.12.21.
+//
+
+
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift:8]: Useless line

File: SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import Foundation
+
+protocol URLBuilderType {
+    func build(from urlString: String) throws -> SearchUrl
+}
+
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift:14]: Useless line

File: SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    
+    // MARK: - Build
+    
+    func build(from urlString: String) throws -> SearchUrl  {
+        guard let url = URL(string: urlString) else {
+            throw URLBuilderError.unableToParseURL
+        }
+        var components = URLComponents(url: url,
+                                       resolvingAgainstBaseURL: true)
+//        let query = apiQuery()
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift:27]: Why keeping dead code?

File: SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+//        var value: String = ""
+//        if let filePath = Bundle.main.path(forResource: "ApiKey", ofType: "plist") {
+//            let plist = NSDictionary(contentsOfFile:filePath)
+//            value = plist?.object(forKey: keyname) as! String
+//        }
+//        return value
+//    }
+}
+
+final class SearchUrl {
```
Comment (Asavarkhul) [inline SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift:54]: Why not using the `Endpoint` protocol?

File: SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift (hunk 2 above)
Comment (PWiley) [reply SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift]: Corrected

File: SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift (hunk 1 above)
Comment (PWiley) [reply SoundsBinder/SoundsBinder/Sources/Network/HTTPClient.swift]: Corrected

File: SoundsBinder/SoundsBinder/Sources/Network/HTTPEngine.swift (hunk 3 above)
Comment (PWiley) [reply SoundsBinder/SoundsBinder/Sources/Network/HTTPEngine.swift]: Corrected

File: SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift (hunk 5 above)
Comment (PWiley) [reply SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift]: Corrected

File: SoundsBinder/SoundsBinder/Sources/Network/RequestCancellationToken.swift (hunk 4 above)
Comment (PWiley) [reply SoundsBinder/SoundsBinder/Sources/Network/RequestCancellationToken.swift]: Corrected

File: SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift (hunk 6 above)
Comment (PWiley) [reply SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift:13]: Corrected

File: SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift (hunk 7 above)
Comment (PWiley) [reply SoundsBinder/SoundsBinder/Sources/Network/URLBuilder.swift]: Corrected

**Pull request closed without merging**